src/openrct2/world/Banner.cpp

```diff
@@ -209,15 +209,21 @@ WallElement* banner_get_scrolling_wall_tile_element(BannerIndex bannerIndex)
  *
  *  rct2: 0x006B7D86
  */
-uint8_t banner_get_closest_ride_index(int32_t x, int32_t y, int32_t z)
+uint8_t banner_get_closest_ride_index(const CoordsXYZ& mapPos)
 {
-    static constexpr const LocationXY16 NeighbourCheckOrder[] = { { 32, 0 },    { -32, 0 },   { 0, 32 },
-                                                                  { 0, -32 },   { -32, +32 }, { +32, -32 },
-                                                                  { +32, +32 }, { -32, +32 }, { 0, 0 } };
-
-    for (size_t i = 0; i < std::size(NeighbourCheckOrder); i++)
+    static constexpr const std::array<CoordsXY, 9> NeighbourCheckOrder = { CoordsXY{ COORDS_XY_STEP, 0 },
```

Comment:
Why an std::array, rather than keeping it as a C array? To me, it seems like the latter is more efficient.

Also, _if_ we decide to stick with std::array, then you should also modify its counterpart in TestPaint.

Reply:
Std::array let's us use the ranged for loop and is no more or less efficient.

Reply:
`std::array` is just a wrapper for C array that is STL compliant and provides convenient functions and usage, such as the one mentioned by @duncanspumpkin

Reply:
Unresolving this - you haven't modified its counterpart in Testpaint yet, which is why the CI is failing.

Reply:
This is just a fancy array rearranged of CoordsDirectionDelta (Except it also checks tile 0). It doesn't actually matter the order that much. Is there a way we could define this as CoordsDirectionDelta + 1 coords for the tile 0?

Reply:
It's possible, but it would be something like (in a simplified example):
```c++
    static constexpr std::array<int, 8> a {1,2,3,4,5,6,7,8};
    static constexpr std::array<int, 9> b {a[0], a[2], a[3], a[4], a[5], a[6], a[7], 0};
```

Reply:
leave as is.